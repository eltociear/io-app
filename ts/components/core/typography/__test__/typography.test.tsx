import React from "react";
import TestRenderer from "react-test-renderer";
import type { IOColors } from "@pagopa/io-app-design-system";
import { IOFontWeight } from "../../fonts";
import { calculateWeightColor } from "../common";
import { LabelSmall } from "../LabelSmall";

describe("Test Typography Components", () => {
  it("LabelSmall Snapshot", () => {
    const labelSmallDefault = TestRenderer.create(
      <LabelSmall>Text</LabelSmall>
    ).toJSON();
    expect(labelSmallDefault).toMatchSnapshot();

    type BodyColors = React.ComponentProps<typeof LabelSmall>["color"];

    const allowedColors: ReadonlyArray<BodyColors> = [
      "blue",
      "bluegrey",
      "red",
      "white"
    ];

    allowedColors.map(color => {
      const labelSmall = TestRenderer.create(
        <LabelSmall color={color}>Text</LabelSmall>
      ).toJSON();
      expect(labelSmall).toMatchSnapshot();
    });
  });
});

describe("Test Typography common", () => {
  it("Test calculateWeightColor behaviour", () => {
    const noValues = calculateWeightColor<IOFontWeight, IOColors>(
      "Bold",
      "red"
    );
    expect(noValues.color).toBe("red");
    expect(noValues.weight).toBe("Bold");

    const weightProvided = calculateWeightColor<IOFontWeight, IOColors>(
      "Bold",
      "red",
      "Regular"
    );
    expect(weightProvided.color).toBe("red");
    expect(weightProvided.weight).toBe("Regular");

    const allValuesProvided = calculateWeightColor<IOFontWeight, IOColors>(
      "Bold",
      "red",
      "Regular",
      "bluegrey"
    );
    expect(allValuesProvided.color).toBe("bluegrey");
    expect(allValuesProvided.weight).toBe("Regular");
  });
});
