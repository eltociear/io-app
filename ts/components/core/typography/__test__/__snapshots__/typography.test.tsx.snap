// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Typography Components Label Snapshot 1`] = `
<Text
  color="blue"
  defaultColor="blue"
  defaultWeight="Bold"
  font="TitilliumSansPro"
  fontStyle={
    {
      "fontSize": 16,
    }
  }
  style={
    [
      {
        "fontSize": 16,
      },
      {
        "color": "#0073E6",
        "fontFamily": "Titillium Sans Pro",
        "fontStyle": "normal",
        "fontWeight": "700",
      },
    ]
  }
  weight="Bold"
>
  Text
</Text>
`;

exports[`Test Typography Components Label Snapshot 2`] = `
<Text
  color="blue"
  defaultColor="blue"
  defaultWeight="Bold"
  font="TitilliumSansPro"
  fontStyle={
    {
      "fontSize": 16,
    }
  }
  style={
    [
      {
        "fontSize": 16,
      },
      {
        "color": "#0073E6",
        "fontFamily": "Titillium Sans Pro",
        "fontStyle": "normal",
        "fontWeight": "700",
      },
    ]
  }
  weight="Bold"
>
  Text
</Text>
`;

exports[`Test Typography Components Label Snapshot 3`] = `
<Text
  color="bluegrey"
  defaultColor="blue"
  defaultWeight="Bold"
  font="TitilliumSansPro"
  fontStyle={
    {
      "fontSize": 16,
    }
  }
  style={
    [
      {
        "fontSize": 16,
      },
      {
        "color": "#475A6D",
        "fontFamily": "Titillium Sans Pro",
        "fontStyle": "normal",
        "fontWeight": "700",
      },
    ]
  }
  weight="Bold"
>
  Text
</Text>
`;

exports[`Test Typography Components Label Snapshot 4`] = `
<Text
  color="white"
  defaultColor="blue"
  defaultWeight="Bold"
  font="TitilliumSansPro"
  fontStyle={
    {
      "fontSize": 16,
    }
  }
  style={
    [
      {
        "fontSize": 16,
      },
      {
        "color": "#FFFFFF",
        "fontFamily": "Titillium Sans Pro",
        "fontStyle": "normal",
        "fontWeight": "700",
      },
    ]
  }
  weight="Bold"
>
  Text
</Text>
`;

exports[`Test Typography Components LabelSmall Snapshot 1`] = `
<Text
  color="blue"
  defaultColor="blue"
  defaultWeight="Bold"
  font="TitilliumSansPro"
  fontStyle={
    {
      "fontSize": 14,
      "lineHeight": 21,
    }
  }
  style={
    [
      {
        "fontSize": 14,
        "lineHeight": 21,
      },
      {
        "color": "#0073E6",
        "fontFamily": "Titillium Sans Pro",
        "fontStyle": "normal",
        "fontWeight": "700",
      },
    ]
  }
  weight="Bold"
>
  Text
</Text>
`;

exports[`Test Typography Components LabelSmall Snapshot 2`] = `
<Text
  color="blue"
  defaultColor="blue"
  defaultWeight="Bold"
  font="TitilliumSansPro"
  fontStyle={
    {
      "fontSize": 14,
      "lineHeight": 21,
    }
  }
  style={
    [
      {
        "fontSize": 14,
        "lineHeight": 21,
      },
      {
        "color": "#0073E6",
        "fontFamily": "Titillium Sans Pro",
        "fontStyle": "normal",
        "fontWeight": "700",
      },
    ]
  }
  weight="Bold"
>
  Text
</Text>
`;

exports[`Test Typography Components LabelSmall Snapshot 3`] = `
<Text
  color="bluegrey"
  defaultColor="blue"
  defaultWeight="Bold"
  font="TitilliumSansPro"
  fontStyle={
    {
      "fontSize": 14,
      "lineHeight": 21,
    }
  }
  style={
    [
      {
        "fontSize": 14,
        "lineHeight": 21,
      },
      {
        "color": "#475A6D",
        "fontFamily": "Titillium Sans Pro",
        "fontStyle": "normal",
        "fontWeight": "700",
      },
    ]
  }
  weight="Bold"
>
  Text
</Text>
`;

exports[`Test Typography Components LabelSmall Snapshot 4`] = `
<Text
  color="red"
  defaultColor="blue"
  defaultWeight="Bold"
  font="TitilliumSansPro"
  fontStyle={
    {
      "fontSize": 14,
      "lineHeight": 21,
    }
  }
  style={
    [
      {
        "fontSize": 14,
        "lineHeight": 21,
      },
      {
        "color": "#C02927",
        "fontFamily": "Titillium Sans Pro",
        "fontStyle": "normal",
        "fontWeight": "700",
      },
    ]
  }
  weight="Bold"
>
  Text
</Text>
`;

exports[`Test Typography Components LabelSmall Snapshot 5`] = `
<Text
  color="white"
  defaultColor="blue"
  defaultWeight="Bold"
  font="TitilliumSansPro"
  fontStyle={
    {
      "fontSize": 14,
      "lineHeight": 21,
    }
  }
  style={
    [
      {
        "fontSize": 14,
        "lineHeight": 21,
      },
      {
        "color": "#FFFFFF",
        "fontFamily": "Titillium Sans Pro",
        "fontStyle": "normal",
        "fontWeight": "700",
      },
    ]
  }
  weight="Bold"
>
  Text
</Text>
`;

exports[`Test Typography Components Link Snapshot 1`] = `
<Text
  color="blue"
  defaultColor="blue"
  defaultWeight="Semibold"
  font="TitilliumSansPro"
  fontStyle={
    {
      "fontSize": 16,
      "textDecorationLine": "underline",
    }
  }
  style={
    [
      {
        "fontSize": 16,
        "textDecorationLine": "underline",
      },
      {
        "color": "#0073E6",
        "fontFamily": "Titillium Sans Pro",
        "fontStyle": "normal",
        "fontWeight": "600",
      },
    ]
  }
  weight="Semibold"
>
  Text
</Text>
`;

exports[`Test Typography Components Monospace Snapshot 1`] = `
<Text
  color="bluegrey"
  defaultColor="bluegrey"
  defaultWeight="Regular"
  font="RobotoMono"
  fontStyle={
    {
      "fontSize": 16,
    }
  }
  style={
    [
      {
        "fontSize": 16,
      },
      {
        "color": "#475A6D",
        "fontFamily": "Roboto Mono",
        "fontStyle": "normal",
        "fontWeight": "400",
      },
    ]
  }
  weight="Regular"
>
  Text
</Text>
`;
